client/dfclient/src/websocket.cpp

```diff
@@ -194,42 +195,40 @@ int WebSocketBeast::Connect(std::string& fullAddress) {
 		return -1;
 	}
 
-	webSocketManager._ws->toBeOpened = true;
+	GlobalWebsocket->toBeOpened = true;
 
 	DoRead();
 
 	new std::thread([&](){
 		try {
 			wsb->ctx.ioc.run();
-		} catch (const std::exception& e) {
-			if (e.what() == std::string("Operation canceled")) {
-				std::cout << "ioc quitting\n";
-				return;
-			}
-			throw e;
+		} catch (const boost::exception& e) {
+			std::cout << "wsb->ctx.ioc.run() failed: " << boost::diagnostic_information(e) << "\n";
 		}
 		std::cout << "ioc quitting\n";
 	});
 
 	return 0;
 }
 
-bool WebSocketBeast::Send(std::string& data) {
-	WebSocketBeast* wsb = (WebSocketBeast*) webSocketManager._ws.get();
+bool WebSocketBeast::Send(const std::string& data) {
+	WebSocketBeast* wsb = (WebSocketBeast*) GlobalWebsocket.get();
 	wsb->ctx.ws.write(net::buffer(std::string(data)));
 }
 
 void WebSocketBeast::Close() {
-	WebSocketBeast* wsb = (WebSocketBeast*) webSocketManager._ws.get();
+	WebSocketBeast* wsb = (WebSocketBeast*) GlobalWebsocket.get();
 	try {
-		wsb->ctx.ws.close(websocket::close_code::normal);
+    	wsb->ctx.ws.close(websocket::close_code::normal);
```

Comment:
Nit: this line is indented with spaces, while the rest is indented with tabs.

Reply:
nice catch